zkevm-circuits/src/copy_circuit.rs

```diff
@@ -625,31 +566,6 @@ impl<F: Field> SubCircuitConfig<F> for CopyCircuitConfig<F> {
                 },
             );
 
-            let value_or_pad_2 = meta.query_advice(value, Rotation(2))
-                * not::expr(meta.query_advice(is_pad, Rotation(2)));
-
-            // TODO: check value_acc(0) = value when is_first
-            cb.require_equal(
-                "value_acc is same for read-write rows",
-                meta.query_advice(value_acc, Rotation::cur()),
-                meta.query_advice(value_acc, Rotation::next()),
-            );
-            cb.condition(
-                and::expr([
-                    not::expr(meta.query_advice(is_last, Rotation::next())),
-                    not::expr(meta.query_advice(is_pad, Rotation::cur())),
-                    not::expr(meta.query_advice(mask, Rotation(2))),
-                ]),
-                |cb| {
-                    cb.require_equal(
-                        "value_acc(2) == value_acc(0) * r + value(2)",
```

Comment:
This is moved above into the `is_continue` block because it always happens, not sometimes.